### PR TITLE
Heretics update part 3

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR
made heretics much less murderboney by adding **living heart**!!
it randomly selects targets for you at the beginning **or** if you invoke the Relentless Heartbeat ritual
image

it allows you to pinpoint your targets and also plays a very audible heartbeat for crewmembers with their volume set on 100 (couldn't be me)
image

and now the rune will refuse any sacrifice other than your targets
image

also fixed some masks not working with madmask ritual and other actions related stuff
and did other stuff READ CL

## Why / Balance
it's the way god intended

:cl:
- add: Added the Living Heart to heretic's arsenal.
- tweak: Eldritch gods will not accept anyone as a valid sacrifice for heretics anymore.
- fix: Fixed some heretics abilities bugs.
- tweak: Made ash heretic's volcano blast ability not suck
- tweak: Updated heretic guidebook GO READ IT!!!

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `LivingHeartMenu` UI component for heretic-related actions.
	- Added functionality for heretics to interact with and activate the Living Heart ability.
	- Enhanced the Heretic abilities with new event handling and user feedback.
	- New rituals and abilities, including "Living Heart" and "Relentless Heartbeat," providing strategic options for players.
	- Added new audio files (`blind.ogg`, `castsummon.ogg`, `disintegrate.ogg`, `hereticknock.ogg`, `repulse.ogg`, `voidblink.ogg`, `heartbeat.ogg`) and metadata for improved ambience related to heretic actions.

- **Bug Fixes**
	- Improved control flow in surgery actions to prevent execution if canceled.
	- Streamlined ritual execution by removing unnecessary checks.

- **Refactor**
	- Updated logic in the `HereticKnowledgeSystem` to ensure knowledge progression aligns with the heretic's current path.
	- Enhanced content organization in ritual and knowledge files for clarity.

- **Chores**
	- Added new dependencies to enhance heretic functionalities across various systems.
	- Updated localization files with new strings for heretic abilities.
	- Included licensing information and sources for new audio files.

### CODE REVIEW

File: Content.Client/_Goobstation/Heretic/UI/LivingHeartMenu.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            var castChild = child as EmbeddedEntityMenuButton;
+            if (castChild == null)
+                continue;
+
+            castChild.OnButtonUp += _ =>
+            {
+                SendActivateMessageAction?.Invoke(castChild.NetEntity);
+                Close();
+            };
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Client/_Goobstation/Heretic/UI/LivingHeartMenu.xaml.cs:88]: see the logic in adding buttons. it only gets invoked once anyways

File: Content.Client/_Goobstation/Heretic/UI/LivingHeartMenu.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            if (target == null) continue;
+
+            var ent = _ent.GetEntity(target);
+            if (ent == null) continue;
+
+            var button = new EmbeddedEntityMenuButton
+            {
+                StyleClasses = { "RadialMenuButton" },
+                SetSize = new Vector2(64, 64),
+                ToolTip = _ent.GetComponent<MetaDataComponent>(ent.Value).EntityName,
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Client/_Goobstation/Heretic/UI/LivingHeartMenu.xaml.cs:54]: said entity must have a metadata component else it's not an entity anymore.
and the null check for the entity itself already exists

File: Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs [hunk 3]
```diff
@@ -133,4 +190,44 @@
...
+    #endregion
+
+
+
+    #region Miscellaneous
+
+    private void OnMagicItemExamine(Entity<HereticMagicItemComponent> ent, ref ExaminedEvent args)
+    {
+        if (!HasComp<HereticComponent>(args.Examiner))
+            return;
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs:227]: args.examiner is not nullable

File: Content.Server/_Goobstation/Heretic/Abilities/HereticAbilitySystem.cs [hunk 4]
```diff
@@ -160,6 +166,59 @@
...
+
+        var isOnStation = targetStation != null && targetStation == ownStation;
+
+        var globalVector = _transform.GetWorldPosition((EntityUid) target, xquery) - _transform.GetWorldPosition(ent, xquery);
+        var globalDirection = ContentLocalizationManager.FormatDirection(globalVector.GetDir());
+
+        var localVector = Transform((EntityUid) target).LocalPosition - Transform(ent).LocalPosition;
+        var localDirection = ContentLocalizationManager.FormatDirection(localVector.GetDir());
+
+        var locstate = state.ToString().ToLower();
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Server/_Goobstation/Heretic/Abilities/HereticAbilitySystem.cs:211]: you can't see localization files therefore you can't make up problems with localization
disregard

File: Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs [hunk 5]
```diff
@@ -77,34 +95,73 @@ private void OnCompInit(Entity<HereticComponent> ent, ref ComponentInit args)
...
+
+        // pick one command staff
+        predicates.Add(t => HasComp<CommandStaffComponent>(t));
+
+        // p[ick one secoff
+        predicates.Add(t =>
+            _prot.TryIndex<DepartmentPrototype>("Security", out var dept) // can we get sec jobs?
+            && _mind.TryGetMind(t, out var mindid, out _) // does it have a mind?
+            && TryComp<JobComponent>(mindid, out var jobc) && jobc.Prototype.HasValue // does it have a job?
+            && dept.Roles.Contains(jobc.Prototype!.Value)); // is that job being shitsec?
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs:131]: `mindid.CurrentJob` does not exist

File: Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs [hunk 6]
```diff
@@ -133,4 +190,44 @@
...
+    private void OnDamage(Entity<HereticComponent> ent, ref DamageModifyEvent args)
+    {
+        if (!ent.Comp.Ascended)
+            return;
+
+        switch (ent.Comp.CurrentPath)
+        {
+            case "Ash":
+                // nullify heat damage because zased
+                args.Damage.DamageDict["Heat"] = 0;
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs:213]: damagetypeclasses does not exist

File: Content.Client/_Goobstation/Heretic/UI/LivingHeartMenu.xaml.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Client/_Goobstation/Heretic/UI/LivingHeartMenu.xaml.cs:54]: disregard my previous message this is actually a good advice

**Review by Piras314 — CHANGES_REQUESTED** [review]
I feel like I'm reading the code a serial killer wrote, why does the livingheartmenu have to compile in it's own goddamn itemgroup in the client csproj? I just realised there's no goddamn copyright info on the heretic sounds at all either. Why aren't there parent base entities for different types of rituals... I did not review heretics well. Also the living heart sprite kinda sucks especially how small it is. The actual ability looks OK

**Comment by whateverusername0** [comment]
@Piras314 

> why does the livingheartmenu have to compile in it's own goddamn itemgroup in the client csproj?

idk i think i forgot to remove that

> I just realised there's no goddamn copyright info on the heretic sounds at all either.

my bad og

> Why aren't there parent base entities for different types of rituals...

do you really need them?

> Also the living heart sprite kinda sucks especially how small it is.

make me a better sprite then :goberserk:

**Comment by whateverusername0** [comment]
> I feel like I'm reading the code a serial killer wrote

:godmode:

File: Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs [hunk 7]
```diff
@@ -77,34 +96,85 @@ private void OnCompInit(Entity<HereticComponent> ent, ref ComponentInit args)
...
     {
-        // ignore damage from heretic stuff
-        if (args.Origin.HasValue && HasComp<HereticBladeComponent>(args.Origin))
-            args.Cancelled = true;
+        foreach (var target in ent.Comp.SacrificeTargets)
+        {
+            // check if sacrificed or whatnot
+            if (!TryGetEntity(target, out var tent) || !Exists(tent))
+                ent.Comp.SacrificeTargets.Remove(target);
+        }
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs:112]: works

**Review by Piras314 — APPROVED** [review]

**Pull request merged**